native/src/main/java/io/ballerina/stdlib/protobuf/nativeimpl/AnyTypeCreator.java

```diff
@@ -58,37 +63,69 @@ public static BString externGetNameFromRecord(BMap<BString, Object> value) {
     }
 
     public static Object unpack(BMap<BString, Object> value, BTypedesc targetType) {
+        typesMap.put(WRAPPER_DOUBLE_TYPE_NAME, TypeTags.FLOAT_TAG);
+        typesMap.put(WRAPPER_FLOAT_TYPE_NAME, TypeTags.FLOAT_TAG);
+
+        typesMap.put(WRAPPER_INT64_TYPE_NAME, TypeTags.INT_TAG);
+        typesMap.put(WRAPPER_UINT64_TYPE_NAME, TypeTags.INT_TAG);
+        typesMap.put(WRAPPER_INT32_TYPE_NAME, TypeTags.INT_TAG);
+        typesMap.put(WRAPPER_UINT32_TYPE_NAME, TypeTags.INT_TAG);
+
+        typesMap.put(WRAPPER_BOOL_TYPE_NAME, TypeTags.BOOLEAN_TAG);
+        typesMap.put(WRAPPER_STRING_TYPE_NAME, TypeTags.STRING_TAG);
+        typesMap.put(WRAPPER_BYTES_TYPE_NAME, TypeTags.ARRAY_TAG);
+        typesMap.put(EMPTY_TYPE_NAME, TypeTags.NULL_TAG);
+        typesMap.put(TIMESTAMP_TYPE_NAME, TypeTags.INTERSECTION_TAG);
+        typesMap.put(DURATION_TYPE_NAME, TypeTags.DECIMAL_TAG);
```

Comment:
Logic wise LGTM. If we could prevent creating this map for each call, it would be much better. But I don’t see any straightforward solution ATM. @MadhukaHarith92 any idea?.

Reply:
+1, moved the map creation logic to protobuf module initialization. Therefore, the map will get populated only once. Fixed in 8c771c1.